public/app/core/components/AppChrome/AppChromeService.tsx

```diff
@@ -94,6 +94,9 @@ export class AppChromeService {
   }
 
   public setReturnToPrevious = (returnToPrevious: ReturnToPreviousProps) => {
+    if (!returnToPrevious.href) {
```

Comment:
here (line 96) is where the `href` prop should be marked as optional. maybe if you export `ReturnToPreviousData` from `@grafana/runtime`, you can do something like:

```tsx
public setReturnToPrevious = (returnToPrevious: ReturnToPreviousData) => {
```

Reply:
actually ignore this, josh's advice about putting that logic in the hook is better 👍